### PR TITLE
*: Add a config for sql parser to enable parsing window function syntax

### PR DESCRIPTION
**Pull request by shenli**
Some keywork for window function make be not compatiable with user's application. 
For example, the `GROUPS` keyword is used as identifier in mybatis's unit test.
So we should disable it by default and enable it by need.

### CODE REVIEW

File: parser/lexer.go [hunk 1]
```diff
@@ -44,6 +44,11 @@ type Scanner struct {
 	specialComment specialCommentScanner
 
 	sqlMode mysql.SQLMode
+
+	// If the lexcer should reconginze keywords for window function.
```
Comment (lamxTyler) [inline parser/lexer.go]: Typo for `lexcer`.

**Comment by wangyisong1996** [comment]
LGTM

**Comment by shenli** [comment]
We could not pass CI now. I will merge this and rebase the `dev/window-function` branch.

**Pull request merged**